tests/virt_autotest/login_console.pm

```diff
@@ -44,19 +45,29 @@ sub double_check_xen_role {
         die 'Double-check xen kernel failed';
     }
 
+    # for modular libvirt, virtxend is expected in "loaded: active or inactive" status.
+    # virtxend.socket seems to be always in "loaded: active" status
+    unless (is_monolithic_libvirtd) {
+        die 'virtxend.socket is not running!' unless script_run("systemctl is-active virtxend.socket") eq 0;
+    }
+
     record_info 'INFO', 'Check if start bootloader from a read-only snapshot';
     assert_script_run('touch /root/read-only.fs && rm -rf /root/read-only.fs');
     save_screenshot;
 }
 
 sub check_kvm_modules {
-    if (script_run('lsmod | grep "^kvm\b"') == 0 and script_run('lsmod | grep -e "^kvm_intel\b" -e "^kvm_amd\b"') == 0) {
+    unless (script_run('lsmod | grep "^kvm\b"') == 0 or script_run('lsmod | grep -e "^kvm_intel\b" -e "^kvm_amd\b"') == 0) {
         save_screenshot;
-        record_info("KVM", "kvm and kvm_intel/amd modules are loaded");
-    }
-    else {
         die "KVM modules are not loaded!";
     }
+
+    # for modular libvirt, virtqemud is expected in "loaded: active or inactive" status.
+    # virtqemud.socket seems to be always in "loaded: active" status
+    unless (is_monolithic_libvirtd) {
+        die 'virtqemud.socket is not running!' unless script_run("systemctl is-active virtqemud.socket") eq 0;
+    }
```

Comment:
This can also be simplified as:
```
die 'virtqemud.socket is not running!' if (script_run("systemctl is-active virtqemud.socket") ne 0 and !is_monolithic_libvirtd)
```